airsonic-main/src/main/java/org/airsonic/player/service/TranscodingService.java

```diff
@@ -202,15 +202,32 @@ public Parameters getParameters(MediaFile mediaFile, Player player, Integer maxB
 
         boolean hls = videoTranscodingSettings != null && videoTranscodingSettings.isHls();
         Transcoding transcoding = getTranscoding(mediaFile, player, preferredTargetFormat, hls);
-        if (transcoding != null) {
-            parameters.setTranscoding(transcoding);
+        Integer bitRate = mediaFile.getBitRate();
+
+        if (mediaFile.isVideo()) {
+            if (bitRate == null) {
+                bitRate = 20000;
+            }
             if (maxBitRate == null) {
-                maxBitRate = mediaFile.isVideo() ? VideoPlayerController.DEFAULT_BIT_RATE : TranscodeScheme.MAX_192.getMaxBitRate();
+                maxBitRate = VideoPlayerController.DEFAULT_BIT_RATE;
+            }
+        } else {
+            if (bitRate == null) {
+                bitRate = TranscodeScheme.MAX_320.getMaxBitRate();
+            } else if (mediaFile.isVariableBitRate()) {
+                // Assume VBR needs approx 20% more bandwidth to maintain equivalent quality in CBR/ABR
+                bitRate = Math.min(bitRate * 6/5, 320);
```

Comment:
I'm not following on the logic here. Can you please expand?

Reply:
I'm trying to preserve as much quality as possible when transcoding (unless the bitrate is limited by other settings).  Since VBR files can encode higher quality content at a lower average bitrate, I assume that a CBR/ABR transcode needs 20% more bandwidth to make sure that complex sections of audio don't run out of bits and end up lower quality.  But there's also an upper limit of 320kbps.

Granted the 20% figure is still just an approximation, but I based it on the bitrate comparisons here: https://trac.ffmpeg.org/wiki/Encode/MP3